daemon/cmd/daemon.go

```diff
@@ -591,6 +591,12 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 	if err != nil {
 		log.WithError(err).Error("Unable to read existing endpoints")
 	}
+	// Restore all proxy ports from datapath, if possible
+	// Must be run before d.bootstrapFQDN(), which depends
+	// on the ports having been restored.
+	if d.l7Proxy != nil {
+		d.l7Proxy.RestoreProxyPorts()
```

Comment:
i'd recommend to guard this check with `option.Config.EnableL7Proxy` instead of the nil check as we use this in all other places too (even though here we shouldn't have the "nil-check with interface issues".

Reply:
I'll have to let this slide this time... Thanks for the review!